python/GafferUI/SpreadsheetUI/_ClipboardAlgo.py

```diff
@@ -334,6 +330,25 @@ def _set( cls, data, plug, atTime ) :
 			for childName, childData in data.items() :
 				cls._set( childData, plug[ childName ], atTime )
 
+	@staticmethod
+	def _canSetKeyOrValue( plug ) :
+
+		def settable( p ) :
+			if Gaffer.Animation.isAnimated( p ) :
+				curve = Gaffer.Animation.acquire( p )
+				return not Gaffer.MetadataAlgo.readOnly( curve )
+			else :
+				return p.settable()
+
+		# Ensure we consider child plugs, eg: components of a V3fPlug
+		if Gaffer.MetadataAlgo.readOnly( plug ) or not settable( plug ) :
+			return False
+		for p in Gaffer.Plug.RecursiveRange( plug ) :
+			if Gaffer.MetadataAlgo.getReadOnly( p ) or not settable( p ) :
```

Comment:
No need to call `settable()` here, as that recurses internally anyway.

Reply:
Ahh cool, thanks [removed](https://github.com/GafferHQ/gaffer/pull/4045/commits/3f64d7dad9f9234b9bec6f41ff3e8bcd4ba97c65). Another one for #4032.